### DISCUSSION TITLE
Why does `bevy::render::render_graph::Node::update()` take a `mut World`?

### DISCUSSION THREAD

**Discussion by waywardmonkeys** [bevyengine/bevy#10283] [Q&A]
The docs for `bevy::render::render_graph::Node::update()` say:

> Updates internal node state using the current render [`World`] prior to the run method.

It has the signature `fn update(&mut self, _world: &mut World)`

So while the docs say that it is updating the node's internal state, it does get a mutable `World`. None of the code within the repository uses it mutably (I have a patch that experimentally makes it take `&World` instead), so I'm curious why it is `mut`?

Is it because someone might legitimately want to mutate the world in that function?

Is it because of some historical legacy?

If for some reason, I did want to mutate the world, would that be fine to do there or might that cause some other issue?

**Comment by IceSentry** [comment]
I guess you are right that it's not currently used internally, but the intention of having both `Node::update()` and `Node::run()` is that update has mutable access to itself and the `World` while `run()` doesn't. It's to give more flexibility but the only important part is that `Node::run()` stays readonly.
